File: 05612.py
n = int(input())
m = int(input())

max = m; zero = False
for i in range(n):
    IN, OUT = map(int, input().split())

    m += IN; m -= OUT
    if m < 0:
        zero = True
    if m > max:
        max = m

if zero:
    print(0)
else:
    print(max)

File: 05612.cpp
#include <iostream>
using namespace std;

int main() {
	int n; cin >> n;
	int m; cin >> m;

	int max = m; bool zero = false;
	for (int i = 0; i < n; i++) {
		int in, out; cin >> in >> out;
		
		m += in; m -= out;
		if (m < 0) zero = true;
		if (m > max) max = m;
	}

	if (zero) cout << 0 << endl;
	else cout << max << endl;
}
